lib/src/components/generic/client.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:grpc/grpc_connection_interface.dart';
+
+import '../../gen/common/v1/common.pb.dart';
+import '../../gen/component/generic/v1/generic.pbgrpc.dart';
+import '../../resource/base.dart';
+import '../../utils.dart';
+import 'generic.dart';
+
+class GenericClient extends Generic implements ResourceRPCClient {
+  @override
+  ClientChannelBase channel;
+
+  @override
+  String name;
+
+  @override
+  GenericServiceClient get client => GenericServiceClient(channel);
```

Comment:
(q) this looks to me like a constructor, but is using the syntax of a getter. Am I misunderstanding the use of `get`? Is this a common thing in dart?

Reply:
(spoke with @clintpurser offline, this all makes sense!)

Reply:
Ethan and I talked offline about this. This is getting an instance of `GenericServiceClient` not `GenericClient`. We are doing it this way to not have to update the GenericServiceClient in each client upon the channel changing in the instance of a reconnection.